src/main/java/ch/njol/skript/sections/ConditionalSection.java

```diff
@@ -0,0 +1,188 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.sections;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.lang.util.CodeSection;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@SuppressWarnings("NotNullFieldNotInitialized")
+public class ConditionalSection extends CodeSection {
+
+	static {
+		Skript.registerSection(ConditionalSection.class,
+			"else",
+			"else if <.+>",
+			"[(1¦if)] <.+>");
+	}
+
+	private enum ConditionalType {
+		ELSE, ELSE_IF, IF
+	}
+
+	private ConditionalType type;
+	private Condition condition;
+
+	private Kleenean hadDelayBefore;
+	private Kleenean hasDelayAfter;
+
+	@Override
+	public boolean init(Expression<?>[] exprs,
+						int matchedPattern,
+						Kleenean isDelayed,
+						ParseResult parseResult,
+						SectionNode sectionNode,
+						List<TriggerItem> triggerItems) {
+		type = ConditionalType.values()[matchedPattern];
+		if (type == ConditionalType.IF || type == ConditionalType.ELSE_IF) {
+			String expr = parseResult.regexes.get(0).group();
+			// Don't print a default error if 'if' keyword wasn't provided
+			condition = Condition.parse(expr, parseResult.mark != 0 ? "Can't understand this condition: '" + expr + "'" : null);
+			if (condition == null)
+				return false;
+		}
+
+		ConditionalSection lastIf;
+		if (type != ConditionalType.IF) {
+			lastIf = getIf(triggerItems);
+			if (lastIf == null) {
+				if (type == ConditionalType.ELSE_IF)
+					Skript.error("'else if' has to be placed just after another 'if' or 'else if' section");
+				else
+					Skript.error("'else' has to be placed just after another 'if' or 'else if' section");
+				return false;
+			}
+		} else {
+			lastIf = null;
+		}
+
+		hadDelayBefore = getParser().getHasDelayBefore();
+		loadCode(sectionNode);
+		hasDelayAfter = getParser().getHasDelayBefore();
+		if (hadDelayBefore.isTrue())
+			return true;
+
+		if (type == ConditionalType.ELSE) {
```

Comment:
some comments here could be useful I think

Reply:
I added some comments, but I also changed the code a bit